2.4/root/usr/bin/run-mongod

```diff
@@ -46,8 +32,8 @@ function cleanup() {
     mongo_remove
   fi
   echo "=> Shutting down MongoDB server ..."
-  if [ -f "${MONGODB_PID_FILE}" ]; then
-    kill -2 $(cat ${MONGODB_PID_FILE})
+  if [[ -s ${MONGODB_DATADIR}/mongod.lock ]]; then
+    kill -2 $(cat ${MONGODB_DATADIR}/mongod.lock)
```

Comment:
If the image provides ability to mount config file scripts should avoid getting values from it. (I thought that OpenShift suggested not to do this to avoid bugs for some user weird config files)

MONGODB_PID_FILE is one example of it. The same value is stored in file `mongod.lock` in dbpath. So no reason to get value of MONGODB_PID_FILE from config file.